src/render/Renderer.cpp

```diff
@@ -506,24 +514,21 @@ void CHyprRenderer::renderWindow(CWindow* pWindow, CMonitor* pMonitor, timespec*
             if (WINBB.y < PWSMON->vecPosition.y) {
                 offset.y = (PWSMON->vecPosition.y - WINBB.y) * PROGRESS;
             } else if (WINBB.y + WINBB.height > PWSMON->vecPosition.y + PWSMON->vecSize.y) {
-                offset.y = (WINBB.y + WINBB.width - PWSMON->vecPosition.y - PWSMON->vecSize.y) * PROGRESS;
+                offset.y = (WINBB.y + WINBB.height - PWSMON->vecPosition.y - PWSMON->vecSize.y) * PROGRESS;
             }
         }
 
         renderdata.x += offset.x;
         renderdata.y += offset.y;
+        pWindow->m_vFloatingOffset = offset;
```

Comment:
I'd rather this was done elsewhere, updating props in the renderer is eeeeehhh

Reply:
Do you mean a new anim variable then? Not sure how else this can be done

Reply:
you can update it in a member of CWindow called from somewhere... aaargghhhhh callbacks in animvars would be convenientttt

maybe make a member of CWorkspace like onRenderOffsetUpdate() and call it from the animmgr and input swipe when the render offset is changed. In it, update the offsets of windows

Reply:
wait we do fuckin have update callbacks they're just flawed... gimme sec

Reply:
who the fuck added em ha

Reply:
rebase to 2930c5cb6f5d71dbdfd4d396287f90e19cd41228 and you can attach an update callback to `renderOffset` and update the offsets there. :D

Reply:
added. should be good now

Reply:
wait theres a weird bug. let me check

Reply:
probably good now

Reply:
I think the callback is acting weird and is sometimes giving segfault from `getWorkspaceByID()` https://github.com/hyprwm/Hyprland/pull/5297/files#diff-870052947341ac7c9dc2754f3de429248a55da4b2e36a2523f566f3bb58f40ecR1158 somehow.
No idea how to fix
Meanwhile this should NOT be merged.

Reply:
you mean getWorkspaceByID() crashes?

odd. Check the stacktrace I guess, use gdb, etc

Reply:
yeah gdb is giving me something like
[1.txt](https://github.com/hyprwm/Hyprland/files/14807115/1.txt)

Reply:
somehow w is a nullptr

Reply:
ok i think i fixed it